dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/ApplicationListener.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.openapi20;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.container.service.app.deploy.ApplicationInfo;
+import com.ibm.ws.container.service.state.ApplicationStateListener;
+import com.ibm.ws.container.service.state.StateChangeException;
+
+import io.openliberty.microprofile.openapi20.utils.LoggingUtils;
+
+@Component(service = { ApplicationStateListener.class }, configurationPolicy = ConfigurationPolicy.IGNORE, immediate = true)
+public class ApplicationListener implements ApplicationStateListener {
+
+    private static final TraceComponent tc = Tr.register(ApplicationListener.class);
+    private ApplicationProcessor appProcessor = null;
+
+    /** {@inheritDoc} */
+    @Override
+    public void applicationStarting(ApplicationInfo appInfo) throws StateChangeException {
+        if (appProcessor != null) {
+            try {
+                if (LoggingUtils.isEventEnabled(tc)) {
+                    Tr.event(tc, "Application starting process started: " + appInfo);
+                }
+                appProcessor.addApplication(appInfo);
+                if (LoggingUtils.isEventEnabled(tc)) {
+                    Tr.event(tc, "Application starting process ended: " + appInfo);
+                }
+            } catch (Throwable e) {
+                if (LoggingUtils.isEventEnabled(tc)) {
+                    Tr.event(tc, "Failed to process application: " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void applicationStarted(ApplicationInfo appInfo) throws StateChangeException {}
+
+    /** {@inheritDoc} */
+    @Override
+    public void applicationStopping(ApplicationInfo appInfo) {
+        if (appProcessor != null) {
+            try {
+                ApplicationProcessor.getInstance().removeApplication(appInfo);
```

Comment:
I was trying to modify the `ApplicationProcessor` so that it was not a component, but this appears to be a little more work than I thought.  This is a remnant of the changes that I made as part of that work that I forgot to undo.  I have changed it back now.